pkg/providers/vsphere/vmprovider_vm.go

```diff
@@ -1580,6 +1593,106 @@ func (vs *vSphereVMProvider) vmCreateDoPlacement(
 		return err
 	}
 
+	return processPlacementResult(vmCtx, vcClient, createArgs, *result)
+}
+
+// vmCreateDoPlacementByGroup places the VM from the group's placement result.
+func (vs *vSphereVMProvider) vmCreateDoPlacementByGroup(
+	vmCtx pkgctx.VirtualMachineContext,
+	vcClient *vcclient.Client,
+	createArgs *VMCreateArgs) error {
+
+	result := placement.DoesVMNeedPlacement(vmCtx)
+	if !result.NeedZonePlacement &&
+		!result.NeedHostPlacement &&
+		!result.NeedDatastorePlacement {
+
+		// VM does not require any type of placement, so we can return early.
+		return processPlacementResult(vmCtx, vcClient, createArgs, result)
+	}
+
+	var vmg vmopv1.VirtualMachineGroup
+	if err := vs.k8sClient.Get(
+		vmCtx,
+		ctrlclient.ObjectKey{
+			Namespace: vmCtx.VM.Namespace,
+			Name:      vmCtx.VM.Spec.GroupName,
+		},
+		&vmg); err != nil {
+
+		return fmt.Errorf("failed to get VM Group object: %w", err)
+	}
+
+	var memberStatus vmopv1.VirtualMachineGroupMemberStatus
+	for _, m := range vmg.Status.Members {
+		if m.Kind == vmCtx.VM.Kind && m.Name == vmCtx.VM.Name {
+			memberStatus = m
+			break
+		}
+	}
+
+	if !pkgcnd.IsTrue(
+		&memberStatus,
+		vmopv1.VirtualMachineGroupMemberConditionGroupLinked) {
+
+		return fmt.Errorf("VM is not linked to its group")
+	}
+
+	if !pkgcnd.IsTrue(
+		&memberStatus,
+		vmopv1.VirtualMachineGroupMemberConditionPlacementReady) {
+
+		return fmt.Errorf("VM Group placement is not ready")
+	}
+
+	placementStatus := memberStatus.Placement
+	if placementStatus == nil {
+		return fmt.Errorf("VM Group placement is empty")
+	}
+
+	vmCtx.Logger.V(6).Info("VM Group placement is ready",
+		"placement", placementStatus)
+
+	result.ZoneName = placementStatus.Zone
+
+	result.PoolMoRef = vimtypes.ManagedObjectReference{
+		Type:  "ResourcePool",
+		Value: placementStatus.Pool,
+	}
+
+	result.HostMoRef = &vimtypes.ManagedObjectReference{
+		Type:  "HostSystem",
+		Value: placementStatus.Node,
+	}
+
+	result.Datastores = make([]placement.DatastoreResult, len(placementStatus.Datastores))
+	for i, ds := range placementStatus.Datastores {
+		if val := ds.DiskKey; val != nil {
+			result.Datastores[i].DiskKey = *val
+			result.Datastores[i].ForDisk = true
+		}
+
+		result.Datastores[i].MoRef = vimtypes.ManagedObjectReference{
+			Type:  "Datastore",
```

Comment:
Use the constant [`vimtypes.ManagedObjectDatastore`](https://github.com/vmware/govmomi/blob/main/vim25/types/enum.go#L8433).